### PR TITLE
ci: adjust the task weight algorithm to make ci tasks more uniform

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref #4399

The tests that took the longest are as follows:
```
ok github.com/tikv/pd/tests/server/api 418.766s coverage: 39.2% of statements in ./...
ok github.com/tikv/pd/pkg/core 261.422s coverage: 26.1% of statements in ./...
ok github.com/tikv/pd/server/api 205.389s coverage: 30.5% of statements in ./...
ok github.com/tikv/pd/tests/server/member 175.151s coverage: 11.4% of statements in ./...
ok github.com/tikv/pd/tests/server/cluster 168.862s coverage: 26.9% of statements in ./...
ok github.com/tikv/pd/pkg/mcs/discovery 122.774s coverage: 20.9% of statements in ./...
ok github.com/tikv/pd/pkg/schedule/scatter 90.437s coverage: 20.5% of statements in ./...
ok github.com/tikv/pd/pkg/utils/etcdutil 81.330s coverage: 41.8% of statements in ./...
ok github.com/tikv/pd/tests/server/region_syncer 80.535s coverage: 21.1% of statements in ./...
ok github.com/tikv/pd/pkg/schedule/schedulers 79.579s coverage: 39.4% of statements in ./...
ok github.com/tikv/pd/tests/server/join 78.284s coverage: 9.6% of statements in ./...
ok github.com/tikv/pd/tests/server/join 77.112s coverage: 9.6% of statements in ./...
ok github.com/tikv/pd/tests/server/config 76.615s coverage: 22.5% of statements in ./...
ok github.com/tikv/pd/tests/pdctl/keyspace 73.011s coverage: 20.9% of statements in ./...
ok github.com/tikv/pd/pkg/election 71.484s coverage: 29.9% of statements in ./...
ok github.com/tikv/pd/tests/pdctl/config 70.265s coverage: 25.4% of statements in ./...
```

### What is changed and how does it work?

```commit-message
```

We will sort tasks by weight in descending order, and then calculate the sum of weight.

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: scripts/ci-subtask.sh [hunk 1]
```diff
@@ -29,11 +29,23 @@ else
...
 
+    # Create an associative array to store the weight of each task.
+    declare -A task_weights
+    for t in ${tasks[@]}; do
+        weight $t
+        task_weights[$t]=$?
+    done
+
+    # Sort tasks by weight in descending order.
+    IFS=$'\n' tasks=($(sort -rn <<<"${tasks[*]}"))
```
Comment (rleungx) [inline scripts/ci-subtask.sh:46]: What does IFS mean?

Comment (lhy1024) [reply scripts/ci-subtask.sh:46]: https://unix.stackexchange.com/questions/184863/what-is-the-meaning-of-ifs-n-in-bash-scripting

**Review by rleungx — APPROVED** [review]

File: scripts/ci-subtask.sh (hunk 1 above)
Comment (HuSharp) [reply scripts/ci-subtask.sh:46]: how about just `tasks=($(printf "%s\n" "${tasks[@]}" | sort -rn))`

**Pull request merged**